bob-access/src/extractor.rs

```diff
@@ -40,23 +40,26 @@ impl<T: Extractor> ExtractorExt for T {
 
     fn extract_basic(&self) -> Result<Credentials, Error> {
         let mut builder = prepare_builder(self)?;
-        if let (Some(username), Some(password)) = 
-            (self.get_header("username")?,
-            self.get_header("password")?)
-        {
-            let creds = builder
-                .with_username_password(username, password)
-                .build();
-            Ok(creds)
-        } else {
-            if let Some(node_name) = self.get_header("node_name")? {
+        match (self.get_header("username")?, self.get_header("password")?) {
+            (Some(username), Some(password)) => {
                 let creds = builder
-                    .with_nodename(node_name)
+                    .with_username_password(username, password)
                     .build();
                 Ok(creds)
-            } else {
-                Err(Error::CredentialsNotProvided("missing username or password".into()))
-            }
+            },
+            (None, None) => {
+                if let Some(node_name) = self.get_header("node_name")? {
+                    let creds = builder
+                        .with_nodename(node_name)
+                        .build();
+                    return Ok(creds);
+                }
+                let creds = builder
```

Comment:
Please, add comment here: `Fallback to special "default" user, when no credentials were provided`